tests/test_base.cpp

```diff
@@ -562,7 +580,7 @@ std::string server_node::config_path() const
 	return m_path;
 }
 
-server_config server_node::config() const
+server_config &server_node::config()
```

Comment:
The purpose of code comments and patch description is to simplify its merge and subsequent bug fixing.
Instead of just applying the patch because it contains good description and comments you force me to think like you did when designed this code, which takes time and probably will not be 100% valid and will definitely be forgotten in a few moments, which will not help debugging other problems when the rise.

Code in `test_enable_backend_after_config_change()` gets the last backend and checks whether its initialization fails. I would never ever think that the last backend initialization has to fail. You not only introduced a patch with unknown problem and 'fix' for that unknown problem, you also hide a gem in the tests which forces the last backend fail to initialize. You didn't write a word about design of this new test, you didn't describe what does it do.
No, this is not going to work.